spring-boot-sample-project/src/main/java/com/example/demo/service/UserService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.demo.service;
+
+import com.example.demo.pojo.User;
+import com.example.demo.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+@Component
+public class UserService {
+
+
+    private UserRepository userRepository = new UserRepository();
+
+    public List<User> getAllUser(){
+        List<User> userList = userRepository.getUserList();
+        return userList;
+    }
+
+    public User getAUser(String name){
+        List<User> userList = userRepository.getUserList();
+        for(User u : userList){
+            if(u.getFirstName().equalsIgnoreCase(name)){
+                return u;
+            }
+        }
+        return null;
+    }
+
+    public boolean createUser(User user){
+        userRepository.getUserList().add(user);
+        return true;
+    }
+
+    public User updateUser(String userName, User user){
+        List<User> userList = userRepository.getUserList();
```

Comment:
DRY -- Do not Repeat Yourself
You already have a function finding the user for you
`getAUser(name)`
you can check for that user and then update the firstname
Btw, you don't need to remove and add the updated user. Java is all about references. So updating the name should just work